burstphoto/texture/texture.metal

```diff
@@ -321,6 +302,47 @@ kernel void blur_mosaic_texture(texture2d<float, access::read> in_texture [[text
 }
 
 
+kernel void calculate_weight_highlights(texture2d<float, access::read> in_texture [[texture(0)]],
+                                        texture2d<float, access::write> weight_highlights_texture [[texture(1)]],
+                                        constant int& exposure_bias [[buffer(0)]],
+                                        constant float& white_level [[buffer(1)]],
+                                        constant float& black_level_mean [[buffer(2)]],
+                                        constant int& kernel_size [[buffer(3)]],
+                                        uint2 gid [[thread_position_in_grid]]) {
+       
+    // load args
+    int texture_width  = in_texture.get_width();
+    int texture_height = in_texture.get_height();
+    
+    // calculate weight based on exposure bias
+    float const weight_exposure = pow(2.0f, float(exposure_bias/100.0f));
+    
+    // find the maximum intensity in a 5x5 window around the main pixel
+    float pixel_value_max = 0.0f;
+      
+    for (int dy = -kernel_size; dy <= kernel_size; dy++) {
+        int y = gid.y + dy;
+        
+        if (0 <= y && y < texture_height) {
+            for (int dx = -kernel_size; dx <= kernel_size; dx++) {
+                int x = gid.x + dx;
+                
+                if (0 <= x && x < texture_width) {
+                    pixel_value_max = max(pixel_value_max, in_texture.read(uint2(x, y)).r);
+                }
+            }
+        }
+    }
+    
+    pixel_value_max = (pixel_value_max-black_level_mean)*weight_exposure + black_level_mean;
+
+    // ensure smooth blending for pixel values between 0.25 and 0.99 of the white level (based on pixel values before exposure correction)
+    float const weight_highlights = clamp(0.99f/0.74f-1.0f/0.74f*pixel_value_max/white_level, 0.0f, 1.0f);
```

Comment:
I think factoring out the 0.74f would improve readability of this line.

Reply:
In this case, I still prefer the current notation, as both fractions would have very many decimal places.